src/main/CommandLine.cpp

```diff
@@ -299,14 +306,30 @@ maybeEnableInMemoryLedgerMode(Config& config, bool inMemory,
         return nullptr;
     }
 
-    // Adjust configs for in-memory-replay mode
-    config.DATABASE = SecretValue{"sqlite3://:memory:"};
-    config.MODE_STORES_HISTORY = false;
+    // Adjust configs for live in-memory-replay mode
+    // Set database to a small sqlite database used to store minimal data needed
+    // to restore the ledger state
+    if (persistPartially)
+    {
+        config.DATABASE = SecretValue{partialDBForInMemoryMode(config)};
+        config.MODE_STORES_HISTORY = true;
+        // Since this mode still stores historical data (needed to restore
+        // ledger state in certain scenarios), set maintenance to run
+        // aggressively so that we only store a few ledgers worth of data
+        config.AUTOMATIC_MAINTENANCE_PERIOD = std::chrono::seconds(30);
+        config.AUTOMATIC_MAINTENANCE_COUNT = 100;
+        config.DISABLE_XDR_FSYNC = false;
```

Comment:
you should not touch `DISABLE_XDR_FSYNC` here

Reply:
Picked up the latest master, which already addresses this problem but not forcing `DISABLE_XDR_FSYNC`, so I believe this is resolved now.